actionview/lib/action_view/helpers/tags/datalist_field.rb

```diff
@@ -0,0 +1,28 @@
+require "action_view/helpers/tags/placeholderable"
+require "action_view/helpers/text_helper"
+
+module ActionView
+  module Helpers
+    module Tags # :nodoc:
+      class DatalistField < Base # :nodoc:
+        include Placeholderable
+        include TextHelper
+
+        def initialize(object_name, method_name, template_object, option_tags, options = {})
+          @option_tags = option_tags
+
+          super(object_name, method_name, template_object, options)
+        end
+
+        def render
+          add_default_name_and_id(@options)
+          list_name = @object_name.to_s + "_" + @method_name.to_s.pluralize
```

Comment:
maybe this would be more understandable?
```ruby
"#{@object_name}_#{@method_name.to_s.pluralize}"
```

String interpolation is usually better than concatenation